addons/mail/controllers/discuss.py

```diff
@@ -323,17 +323,14 @@ def mail_attachment_delete(self, attachment_id, access_token=None, **kwargs):
     @http.route('/mail/message/add_reaction', methods=['POST'], type='json', auth='public')
     def mail_message_add_reaction(self, message_id, content):
         guest_sudo = request.env['mail.guest']._get_guest_from_request(request).sudo()
-        message_sudo = guest_sudo.env['mail.message'].browse(int(message_id)).exists()
+        message_sudo = guest_sudo.env["mail.message"]._get_message_with_access(message_id, "write")
```

Comment:
I guess we need `_mail_post_access` instead of always `write`, back to writing even more tests and changing code.

Reply:
Actually nevermind, this is already handled in ACL check of message.